### PR TITLE
Dockerised reports with clinic-config and added github actions

### PR DESCRIPTION
**Pull request by kavitha-sundararajan**

### CODE REVIEW

**Comment by gsluthra** [comment]
Please add JIRA issue ID to PR.

**Comment by kavitha-sundararajan** [comment]
This PR is with reference to the card [BAH-1794](https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/39?modal=detail&selectedIssue=BAH-1794). The PRs in the card BAH-1794 are for generating reports image with default-config and they got merged previously. 
The current PR is to incorporate clinic-config in place of default-config and helm charts configurations for the reports.

File: .github/workflows/build_publish_reports.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Docker Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: bahmni-reports/package/docker/Dockerfile
+          push: true
+          tags: bahmniindiadistro/reports:${{env.ARTIFACT_VERSION}},bahmniindiadistro/reports:latest
```
Comment (N0-man) [inline .github/workflows/build_publish_reports.yml:40]: Should be bahmni-reports following the conventions

File: .github/workflows/build_publish_reports.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Helm - Checkout Charts Repository
+        uses: actions/checkout@v2
+        with:
+          repository: Bahmniindiadistro/helm-charts
+          ref: gh-pages
+          path: helm-charts
+          persist-credentials: false
+
+      - name: Helm - Copy chart
+        run: mkdir -p helm-charts/reports/ && cp reports-${{ env.ARTIFACT_VERSION }}.tgz helm-charts/reports/
```
Comment (N0-man) [inline .github/workflows/build_publish_reports.yml:62]: Same, should be bahmni-reports

File: bahmni-reports/package/docker/scripts/docker_build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# Packaging clinic config to embed into default image
+# Working directory : clinic-config
+cd clinc-config && scripts/package.sh && cd ..
+cp clinc-config/target/clinic_config.zip bahmni-india-package/bahmni-reports/resources/
+
+cd bahmni-india-package
+
+# Building Docker images
+REPORTS_IMAGE_TAG=$(cat bahmni-reports/.appversion)-${GITHUB_RUN_NUMBER}
+docker build -t bahmniindiadistro/reports:${REPORTS_IMAGE_TAG} -f bahmni-reports/docker/Dockerfile  . --no-cache
```
Comment (N0-man) [inline bahmni-reports/package/docker/scripts/docker_build.sh:15]: You are using GitHub actions for docker build and push. You don't need this shell

File: bahmni-reports/package/docker/scripts/docker_publish.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+docker push bahmniindiadistro/reports:${REPORTS_IMAGE_TAG}
+
+echo "[INFO] Tagging build images as SNAPSHOT Images"
+REPORTS_SNAPSHOT_IMAGE_TAG=$(cat bahmni-emr/.appversion)-SNAPSHOT
+docker tag bahmniindiadistro/reports:${REPORTS_IMAGE_TAG} bahmniindiadistro/reports:${REPORTS_SNAPSHOT_IMAGE_TAG}
+
+echo "[INFO] Pushing SNAPSHOT images"
+docker push bahmniindiadistro/reports:${REPORTS_SNAPSHOT_IMAGE_TAG}
+
+docker logout
```
Comment (N0-man) [inline bahmni-reports/package/docker/scripts/docker_publish.sh:15]: You are using GitHub actions for docker build and push. You don't need this shell

File: bahmni-reports/package/helm/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: reports
```
Comment (N0-man) [inline bahmni-reports/package/helm/Chart.yaml:2]: bahmni-reports

File: bahmni-reports/package/helm/templates/secrets.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  OPENMRS_DB_HOST: {{ .Values.secrets.OPENMRS_DB_HOST | b64enc | quote }}
+  OPENMRS_DB_USERNAME: {{ .Values.secrets.OPENMRS_DB_USERNAME | b64enc | quote }}
+  OPENMRS_DB_PASSWORD: {{ .Values.secrets.OPENMRS_DB_PASSWORD | b64enc | quote }}
+  REPORTS_DB_SERVER: {{ .Values.secrets.REPORTS_DB_SERVER | b64enc | quote }}
+  REPORTS_DB_USERNAME: {{ .Values.secrets.REPORTS_DB_USERNAME | b64enc | quote }}
+  REPORTS_DB_PASSWORD: {{ .Values.secrets.REPORTS_DB_PASSWORD | b64enc | quote }}
+  OPENELIS_DB_SERVER: {{ .Values.secrets.OPENELIS_DB_SERVER | b64enc | quote }}
+  ODOO_DB_SERVER: {{ .Values.secrets.ODOO_DB_SERVER | b64enc | quote }}
+  ODOO_DB_USERNAME: {{ .Values.secrets.ODOO_DB_USERNAME | b64enc | quote }}
+  ODOO_DB_PASSWORD: {{ .Values.secrets.ODOO_DB_PASSWORD | b64enc | quote }}
```
Comment (N0-man) [inline bahmni-reports/package/helm/templates/secrets.yaml:16]: We won't be running odoo, can we get rid of these env variables

**Review by N0-man — CHANGES_REQUESTED** [review]

File: bahmni-reports/package/helm/templates/secrets.yaml (hunk 6 above)
Comment (kavitha-sundararajan) [reply bahmni-reports/package/helm/templates/secrets.yaml]: removed these variables

File: bahmni-reports/package/docker/scripts/docker_publish.sh (hunk 4 above)
Comment (kavitha-sundararajan) [reply bahmni-reports/package/docker/scripts/docker_publish.sh]: removed the script

File: bahmni-reports/package/docker/scripts/docker_build.sh (hunk 3 above)
Comment (kavitha-sundararajan) [reply bahmni-reports/package/docker/scripts/docker_build.sh]: removed the scrip

File: .github/workflows/build_publish_reports.yml (hunk 1 above)
Comment (kavitha-sundararajan) [reply .github/workflows/build_publish_reports.yml:40]: Renaming of service names will be added as a backlog in ABDM board and picked up later

**Review by N0-man — APPROVED** [review]

**Pull request merged**